### PR TITLE
{AKS} Add an adapter for the cmd object to avoid abuse while verifying the parameter

### PR DESCRIPTION
**Pull request by FumingZhang**
**Description**

* Add an adapter named `AKSCMDAdapter` to limit the scope of use of cmd (with type `AzCliCommand`) provided by cli
* Add a decorator to replace the incoming cmd object of type `AzCliCommand` with an adaptee of type `AKSCMDAdapter`, thereby limiting the abuse of the original object and verifying the validity of the passed object.
* Fix other style issues.
* **Do not merge! Waiting for test!** 

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/_helpers.py [hunk 1]
```diff
@@ -3,13 +3,86 @@
...
+        return self.cmd.supported_api_version(
+            resource_type=resource_type,
+            min_api=min_api,
+            max_api=max_api,
+            operation_group=operation_group,
+            parameter_name=parameter_name,
+        )
+
+
+def AKSCMDDecorator(func):
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/_helpers.py:67]: unit test.

File: src/azure-cli/azure/cli/command_modules/acs/_helpers.py [hunk 2]
```diff
@@ -3,13 +3,86 @@
...
+    # This adapter is used to limit the scope of use of cmd (with type `AzCliCommand`) provided by cli,
+    # and the limitation on the available methods is achieved by using combination.
+    # In addition, after adopting this adapter, it is more convenient to mock the behavior of `cmd` when
+    # writing unit test cases for commands.
+    def __init__(self, cmd, **kwargs):
+        self.cmd = cmd
+        self.validate_methods()
+        self.set_attributes(**kwargs)
+
+    def validate_methods(self):
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/_helpers.py:27]: __validate_methods

File: src/azure-cli/azure/cli/command_modules/acs/_helpers.py [hunk 3]
```diff
@@ -3,13 +3,86 @@
...
+            )
+            raise CLIInternalError(msg)
+        if not callable(getattr(self.cmd, "supported_api_version", None)):
+            msg = (
+                "The given cmd object (type: '{}') has no '{}' method!"
+                "Please check the passed parameter!".format("supported_api_version", type(self.cmd))
+            )
+            raise CLIInternalError(msg)
+
+    def set_attributes(self, **kwargs):
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/_helpers.py:41]: __set_attributes

**Pull request closed without merging**